frontend/src/static/js/v_authorship.js

```diff
@@ -103,9 +106,36 @@ window.vAuthorship = {
       this.totalLineCount = totalLineCount;
       res.sort((a, b) => b.lineCount - a.lineCount);
 
+      this.filesLineInfo = this.convertLineCountToString(filesLineInfoArray);
       this.files = res;
       this.isLoaded = true;
     },
+
+    addLineCountToFileType(path, lineCount, filesLineInfoArray) {
+      const out = {};
+      var fileType = path.split(".").pop();
+      out.fileType = (fileType.length === 0) ? "others" : fileType;
+      out.lineCount = lineCount;
+      var index = filesLineInfoArray.findIndex((x) => x.fileType === out.fileType);
+      if (index === -1) {
+        filesLineInfoArray.push(out);
+      } else {
+        filesLineInfoArray[index].lineCount += out.lineCount;
+      }
+    },
+
+    convertLineCountToString(filesLineInfoArray) {
+      let filesInfoArray = [];
+      for (var key in filesLineInfoArray) {
+        if (Object.prototype.hasOwnProperty.call(filesLineInfoArray, key)) {
+          var fileLineInfo = filesLineInfoArray[key].fileType + ": "
+              + filesLineInfoArray[key].lineCount;
+          filesInfoArray.push(fileLineInfo);
+        }
```

Comment:
use a object then use `Object.keys`, cleaner, eg.

```
filesLineInfo = {
    'java': 100,
    'others': 2
};
```

just push it in, dun need add another `fileLineInfo`. cleaner, something like `filesInfoArray.push(...);`

Reply:
@ongspxm I've made the changes by using a ```filesInfoObj``` and iterating through it using ```Object.keys``` as requested above